python/ray/air/checkpoint.py

```diff
@@ -467,6 +467,9 @@ def from_checkpoint(cls, other: "Checkpoint") -> "Checkpoint":
             >>> model = checkpoint.get_model()  # doctest: +SKIP
             Linear(in_features=1, out_features=1, bias=True)
         """
+        if type(other) is cls:
```

Comment:
this behavior makes sense to me but I need another pair of eyes with full checkpoint context. @krfricke and @xwjiang2010 ?

Reply:
Main question here is if we should be less strict by using `isinstance(other, type)`, which will also return True for inherited classes.